homeassistant/components/mqtt/climate.py

```diff
@@ -372,32 +380,39 @@ class MqttClimate(MqttEntity, ClimateEntity):
     _entity_id_format = climate.ENTITY_ID_FORMAT
     _attributes_extra_blocked = MQTT_CLIMATE_ATTRIBUTES_BLOCKED
 
-    def __init__(self, hass, config, config_entry, discovery_data):
+    _action: HVACAction | None
+    _aux: bool
+    _command_templates: dict[str, Callable[..., PublishPayloadType]]
+    _current_fan_mode: str | None
+    _current_operation: HVACMode | None
+    _current_swing_mode: str | None
+    _feature_preset_mode: bool
+    _optimistic_preset_mode: bool
+    _target_temp: float | None
+    _target_temp_high: float | None
+    _target_temp_low: float | None
+    _topic: dict[str, Any]
+    _value_templates: dict[str, Callable[..., ReceivePayloadType]]
```

Comment:
Instead of declaring all these variables, could you not use `_attr_` on the climate platform (as a preliminary PR)?

Reply:
I'll split the climate part from this PR, and opened: https://github.com/home-assistant/core/pull/81396 (Draft).